lib/EnsEMBL/REST/Model/ga4gh/Beacon.pm

```diff
@@ -362,22 +354,29 @@ sub beacon_query {
     $input_coord{'ref_allele'} = $data->{referenceBases};
     $input_coord{'alt_allele'} = $data->{alternateBases} ? $data->{alternateBases} : $data->{variantType};
 
-    # SNV or small indels have a start
-    # Structural variants can have start-end or startMin/startMax-endMin/endMax
-    $input_coord{'start'} = $data->{start} ? $data->{start} : $data->{startMin};
-    $input_coord{'start_max'} = $data->{startMax} ? $data->{startMax} : $input_coord{'start'};
+    # Types of query:
+    #  1) 'start' with no 'end'; can be done for SNV and small indels
+    #  2) 'start' and 'end' (range query); any variant falling fully or partially within the range
+    #  3) [start1, start2] and [end1, end2] (bracket query); can be used to match any contiguous genomic interval, e.g. for querying imprecise positions
+    my @start_list = split /,/, $data->{start};
+    my @end_list = split /,/, $data->{end} if ($data->{end});
+    my $start_1 = $start_list[0] if($start_list[0]);
+    my $start_2 = $start_list[1] if($start_list[1]);
+    my $end_1 = $end_list[0] if($end_list[0]);
+    my $end_2 = $end_list[1] if($end_list[1]);
+
+    $input_coord{'start'} = $start_1;
+    $input_coord{'start_max'} = $start_2 ? $start_2 : $input_coord{'start'};
 
     my $end;
     if($data->{end}) {
-      $input_coord{'end'} = $data->{end};
+      $input_coord{'end'} = $end_2 ? $end_2 : $end_1;
+      $input_coord{'end_min'} = $end_2 ? $end_1 : $input_coord{'end'};
```

Comment:
It's intended to be `$end_2`. 
If `$end_2` is defined then the min end is supposed to be the only defined end value which is `$end_1`. Otherwise, if `$end_2` is not defined then the min end is the only end that exists `$input_coord{'end'}`.